x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_rule.test.ts

```diff
@@ -208,4 +210,102 @@ describe('buildRule', () => {
     };
     expect(rule).toEqual(expected);
   });
+
+  test('it builds a rule and removes internal tags', () => {
+    const ruleParams = sampleRuleAlertParams();
+    const rule = buildRule({
+      actions: [],
+      doc: sampleDocNoSortId(),
+      ruleParams,
+      name: 'some-name',
+      id: sampleRuleGuid,
+      enabled: false,
+      createdAt: '2020-01-28T15:58:34.810Z',
+      updatedAt: '2020-01-28T15:59:14.004Z',
+      createdBy: 'elastic',
+      updatedBy: 'elastic',
+      interval: 'some interval',
+      tags: [
+        'some fake tag 1',
+        'some fake tag 2',
+        `${INTERNAL_RULE_ID_KEY}:rule-1`,
+        `${INTERNAL_IMMUTABLE_KEY}:true`,
+      ],
+      throttle: 'no_actions',
+    });
+    const expected: Partial<RulesSchema> = {
+      actions: [],
+      author: ['Elastic'],
+      building_block_type: 'default',
+      created_by: 'elastic',
+      description: 'Detecting root and admin users',
+      enabled: false,
+      false_positives: [],
+      from: 'now-6m',
+      id: '04128c15-0d1b-4716-a4c5-46997ac7f3bd',
+      immutable: false,
+      index: ['auditbeat-*', 'filebeat-*', 'packetbeat-*', 'winlogbeat-*'],
+      interval: 'some interval',
+      language: 'kuery',
+      license: 'Elastic License',
+      max_signals: 10000,
+      name: 'some-name',
+      output_index: '.siem-signals',
+      query: 'user.name: root or user.name: admin',
+      references: ['http://google.com'],
+      risk_score: 50,
+      risk_score_mapping: [],
+      rule_id: 'rule-1',
+      severity: 'high',
+      severity_mapping: [],
+      tags: ['some fake tag 1', 'some fake tag 2'],
+      threat: [],
+      to: 'now',
+      type: 'query',
+      note: '',
+      updated_by: 'elastic',
+      updated_at: rule.updated_at,
+      created_at: rule.created_at,
+      throttle: 'no_actions',
+      exceptions_list: getListArrayMock(),
+      version: 1,
+    };
+    expect(rule).toEqual(expected);
+  });
+
```

Comment:
Next 4 tests are moved from `build_signal.test.ts` since `removeInternalTagsFromRule` was moved from `build_signal.ts` to `build_rule.ts`